aws-ts-eks/cluster.ts

```diff
@@ -0,0 +1,457 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import * as k8s from "@pulumi/kubernetes";
+import * as k8sInputs from "@pulumi/kubernetes/types/input";
+
+import { Dashboard } from "./dashboard";
+import { ServiceRole } from "./serviceRole";
+
+/**
+ * EBSVolumeType lists the set of volume types accepted by an EKS storage class.
+ */
+export type EBSVolumeType = "io1" | "gp2" | "sc1" | "st1";
+
+/**
+ * EKSStorageClass describes the inputs to a single Kubernetes StorageClass provisioned by AWS. Any number of storage
+ * classes can be added to a cluster at creation time. One of these storage classes may be configured the default
+ * storage class for the cluster.
+ */
+export interface EKSStorageClass {
+    /**
+     * The EBS volume type.
+     */
+    readonly type: pulumi.Input<EBSVolumeType>;
```

Comment:
Why `readonly`?  Unnecessarily pedantic - and makes it hard to use - i.e., you can no longer do:

```
let x: EKSStorageClass = { ... }
if (something) {
 x.type = "somethingelse";
}
```

(and yes - we should fix this in tfgen as well).

Reply:
Pattern-matching against our generated code :)